tritium-metrics-jvm/src/main/java/com/palantir/tritium/metrics/jvm/JvmMetrics.java

```diff
@@ -190,6 +183,16 @@ protected Ratio getRatio() {
         });
     }
 
+    private static long totalHeapPlusNonHeap(MemoryMXBean memoryBean, ToLongFunction<MemoryUsage> longFunction) {
+        MemoryUsage heap = memoryBean.getHeapMemoryUsage();
+        MemoryUsage nonHeap = memoryBean.getNonHeapMemoryUsage();
+        if (heap == null && nonHeap == null) {
+            return -1;
+        }
+        return (heap == null ? 0 : longFunction.applyAsLong(heap))
+                + (nonHeap == null ? 0 : longFunction.applyAsLong(nonHeap));
```

Comment:
In the case that heap or non-heap is unknown, I'm not sure that we can safely report a total value assuming heap/non-heap doesn't exist. I'd prefer not to report data rather than reporting misleading data.